lib/apiservers/engine/backends/volume.go

```diff
@@ -110,21 +109,24 @@ func (v *Volume) VolumeCreate(name, driverName string, opts, labels map[string]s
 		return result, derr.NewErrorWithStatusCode(fmt.Errorf("Bad Driver Arg: %s", varErr), http.StatusBadRequest)
 	}
 
-	// TODO: setup name randomization if name == nil
+	if model.Name == "" {
+		volumeUUID, _ := uuid.NewUUID()
+		model.Name = volumeUUID.String()
+	}
 
 	res, err := client.Storage.CreateVolume(storage.NewCreateVolumeParams().WithVolumeRequest(model))
 	if err != nil {
 		switch err := err.(type) {
 
 		case *storage.CreateVolumeInternalServerError:
 			// FIXME: right now this does not return an error model...
-			return result, derr.NewErrorWithStatusCode(fmt.Errorf("Server error from Portlayer: %s", err.Error()), http.StatusInternalServerError)
+			return result, derr.NewErrorWithStatusCode(fmt.Errorf("%s", err.Error()), http.StatusInternalServerError)
 
 		case *storage.CreateVolumeDefault:
-			return result, derr.NewErrorWithStatusCode(fmt.Errorf("Server error from Portlayer: %s", err.Payload.Message), http.StatusInternalServerError)
+			return result, derr.NewErrorWithStatusCode(fmt.Errorf("%s", err.Payload.Message), http.StatusInternalServerError)
```

Comment:
I removed the "error from port layer server" since we talked about the error reporting and we don't really want to see `error from daemon server: error from portlayer server :`